saleor/graphql/product/filters.py

```diff
@@ -215,24 +216,24 @@ def filter_products_by_collections(qs, collection_pks):
 
 
 def filter_products_by_stock_availability(qs, stock_availability, channel_slug):
-    total_stock = (
-        Stock.objects.for_channel(channel_slug)
-        .select_related("product_variant")
-        .values("product_variant__product_id")
-        .annotate(
-            total_quantity_allocated=Coalesce(Sum("allocations__quantity_allocated"), 0)
-        )
-        .annotate(total_quantity=Coalesce(Sum("quantity"), 0))
-        .annotate(total_available=F("total_quantity") - F("total_quantity_allocated"))
-        .filter(
-            total_available__lte=0,
+    stocks = Stock.objects.for_channel(channel_slug)
```

Comment:
Could we move this line just before 226. IMO it will be more readable. As I read it, I looked at each line up to 226 to find a use of the `stocks` variable.